src/main/java/com/elastic/support/diagnostics/commands/UserRoleCheckCmd.java

```diff
@@ -0,0 +1,85 @@
+package com.elastic.support.diagnostics.commands;
+
+import com.elastic.support.config.Version;
+import com.elastic.support.diagnostics.chain.Command;
+import com.elastic.support.diagnostics.chain.DiagnosticContext;
+import com.elastic.support.rest.RestClient;
+import com.elastic.support.rest.RestResult;
+import com.elastic.support.util.JsonYamlUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserRoleCheckCmd implements Command {
+
+    Logger logger = LogManager.getLogger(UserRoleCheckCmd.class);
+
+    @Override
+    public void execute(DiagnosticContext context) {
+
+        RestClient restClient = context.getEsRestClient();
+        String user = context.getDiagnosticInputs().getUser();
+
+        Version version = context.getVersion();
+        String url = null;
+        int major = version.getMajor();
+        int minor = version.getMinor();
+
+        if(major == 2){
+            url = "/_shield/user/" + user;
+        }
+        else if (major <= 6){
+            url = "/_xpack/security/user/" + user;
+        }
+        else{
+            url = "/_security/user/" + user;
+        }
+
+        boolean hasAuthorization = false;
+        RestResult result = restClient.execQuery(url);
+        if(result.getStatus() == 200){
+            String userJsonString = result.toString();
+            JsonNode userNode = JsonYamlUtils.createJsonNodeFromString(userJsonString);
+            JsonNode rolesNode = userNode.path("roles");
+            List<String> roles = new ArrayList<>();
+            if(rolesNode.isArray()){
+                ObjectMapper mapper = new ObjectMapper();
+                roles = mapper.convertValue(rolesNode, List.class);
+            }
+
+            if(major < 6){
```

Comment:
This could be simplified to:

```java
// prior to the opening of X-Pack the "superuser" user role necessary was called "admin"
if (major < 6 || (major == 6 && minor < 3)) {
    hasAuthorization = roles.contains("admin");
} else {
    hasAuthorization = roles.contains("superuser");
}
```